freeipa/src/main/java/com/sequenceiq/freeipa/converter/telemetry/TelemetryConverter.java

```diff
@@ -0,0 +1,131 @@
+            telemetry.setFeatures(features);
+            telemetry.setDatabusEndpoint(databusEndpoint);
+            telemetry.setFluentAttributes(request.getFluentAttributes());
+        }
+        return telemetry;
+    }
+
+    public TelemetryResponse convert(Telemetry telemetry) {
+        TelemetryResponse response = null;
+        if (freeIpaTelemetryEnabled && telemetry != null) {
+            response = new TelemetryResponse();
+            response.setFluentAttributes(telemetry.getFluentAttributes());
+            Logging logging = telemetry.getLogging();
+            LoggingResponse loggingResponse = createLoggingResponseFromSource(logging);
+            response.setLogging(loggingResponse);
+            Features features = telemetry.getFeatures();
+            FeaturesResponse featuresResponse = createFeaturesResponseFromSource(features);
+            response.setFeatures(featuresResponse);
+        }
+        return response;
+    }
+
+    private Logging createLoggingFromRequest(LoggingRequest loggingRequest) {
+        Logging logging = null;
+        if (loggingRequest != null) {
+            logging = new Logging();
+            logging.setStorageLocation(loggingRequest.getStorageLocation());
+            if (loggingRequest.getS3() != null) {
+                S3CloudStorageV1Parameters s3Params = new S3CloudStorageV1Parameters();
+                s3Params.setInstanceProfile(loggingRequest.getS3().getInstanceProfile());
+                logging.setS3(s3Params);
+            } else if (loggingRequest.getAdlsGen2() != null) {
+                AdlsGen2CloudStorageV1Parameters adlsGen2Params = new AdlsGen2CloudStorageV1Parameters();
+                adlsGen2Params.setAccountKey(loggingRequest.getAdlsGen2().getAccountKey());
+                adlsGen2Params.setAccountName(loggingRequest.getAdlsGen2().getAccountName());
+                adlsGen2Params.setSecure(loggingRequest.getAdlsGen2().isSecure());
+                adlsGen2Params.setManagedIdentity(logging.getAdlsGen2().getManagedIdentity());
+                logging.setAdlsGen2(adlsGen2Params);
+            }
+        }
+        return logging;
+    }
+
+    private LoggingResponse createLoggingResponseFromSource(Logging logging) {
+        LoggingResponse loggingResponse = null;
+        if (logging != null) {
+            loggingResponse = new LoggingResponse();
+            loggingResponse.setStorageLocation(logging.getStorageLocation());
+            if (logging.getS3() != null) {
+                S3CloudStorageV1Parameters s3Params = new S3CloudStorageV1Parameters();
+                s3Params.setInstanceProfile(logging.getS3().getInstanceProfile());
+                loggingResponse.setS3(s3Params);
+            } else if (logging.getAdlsGen2() != null) {
+                AdlsGen2CloudStorageV1Parameters adlsGen2Params = new AdlsGen2CloudStorageV1Parameters();
+                adlsGen2Params.setAccountKey(logging.getAdlsGen2().getAccountKey());
```

Comment:
Extract `logging.getAdlsGen2()` as local variable and replace all below occurrences.